### PR TITLE
Render icons to render surface

### PR DESCRIPTION
**Pull request by LarsIngo**

### CODE REVIEW

File: src/Engine/Manager/RenderManager.cpp [hunk 1]
```diff
@@ -283,6 +257,63 @@ void RenderManager::Render(World& world, const glm::mat4& viewMatrix, const glm:
...
+    if (cameras) {
+        for (Lens* lens : lenses.GetAll())
+            renderer->RenderIcon(lens->entity->GetWorldPosition(), cameraTexture);
+    }
+
+    renderer->StopRenderingIcons();
+    renderSurface->GetShadingFrameBuffer()->Unbind();
+
+    // Render physics.
+    if (physics) {
```
Comment (NinjaDanz3r) [inline src/Engine/Manager/RenderManager.cpp]: MINOR The function 'GetAnimations' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

File: src/Engine/Manager/RenderManager.cpp [hunk 2]
```diff
@@ -283,6 +257,63 @@ void RenderManager::Render(World& world, const glm::mat4& viewMatrix, const glm:
...
+            renderer->RenderIcon(light->entity->GetWorldPosition(), lightTexture);
+
+        for (PointLight* light : pointLights.GetAll())
+            renderer->RenderIcon(light->entity->GetWorldPosition(), lightTexture);
+
+        for (SpotLight* light : spotLights.GetAll())
+            renderer->RenderIcon(light->entity->GetWorldPosition(), lightTexture);
+    }
+
+    // Render cameras.
```
Comment (NinjaDanz3r) [inline src/Engine/Manager/RenderManager.cpp]: MINOR Variable 'name' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 9 issues
* ![MINOR][MINOR] 9 minor

Watch the comments in this conversation to review them.

#### 7 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L184](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c518da7ebc08e4a0870b99ad1832a2d9adef6df5/src/Engine/Hymn.cpp#L184): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L322](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c518da7ebc08e4a0870b99ad1832a2d9adef6df5/src/Engine/Manager/RenderManager.cpp#L322): The function 'GetDirectionalLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L361](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c518da7ebc08e4a0870b99ad1832a2d9adef6df5/src/Engine/Manager/RenderManager.cpp#L361): The function 'GetMaterials' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L379](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c518da7ebc08e4a0870b99ad1832a2d9adef6df5/src/Engine/Manager/RenderManager.cpp#L379): The function 'GetMeshes' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L398](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c518da7ebc08e4a0870b99ad1832a2d9adef6df5/src/Engine/Manager/RenderManager.cpp#L398): The function 'GetPointLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L419](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/c518da7ebc08e4a0870b99ad1832a2d9adef6df5/src/Engine/Manager/RenderManager.cpp#L419): The function 'GetSpotLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Engine/Manager/RenderManager.cpp [hunk 3]
```diff
@@ -115,10 +134,30 @@ void RenderManager::Render(World& world, Entity* camera) {
 
                     VRDevice* headset = camera->GetComponent<VRDevice>();
 
-                    const glm::mat4 projectionMat = headset->GetHMDProjectionMatrix(nEye, lens->zNear, lens->zFar);
-                    glm::mat4 eyeTranslation = Managers().vrManager->GetHMDHeadToEyeMatrix(nEye);
-
-                    Render(world, glm::inverse(camera->GetModelMatrix()) * eyeTranslation, projectionMat, hmdRenderSurface);
+                    const glm::mat4 lensViewMatrix = glm::inverse(camera->GetModelMatrix());
+                    const glm::mat4 eyeTranslation = Managers().vrManager->GetHMDHeadToEyeMatrix(nEye);
+                    const glm::mat4 eyeViewMatrix = eyeTranslation * lensViewMatrix;
```
Comment (Chainsawkitten) [inline src/Engine/Manager/RenderManager.cpp]: Are we certain this is correct?

File: src/Engine/Manager/DebugDrawingManager.cpp [hunk 4]
```diff
@@ -131,43 +131,33 @@ void DebugDrawingManager::Update(float deltaTime) {
...
+    // Bind render target.
+    renderSurface->GetShadingFrameBuffer()->BindWrite();
+    debugDrawing->StartDebugDrawing(projectionMatrix * viewMatrix);
         
-        // Points.
-        for (const DebugDrawing::Point& point : points) {
-            debugDrawing->DrawPoint(point);
-        }
+    // Points.
+    for (const DebugDrawing::Point& point : points) {
```
Comment (Sajdkick) [inline src/Engine/Manager/DebugDrawingManager.cpp]: Im pretty sure we're not supposed to have {} if there's only one line of code.

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 9 issues
* ![MINOR][MINOR] 9 minor

Watch the comments in this conversation to review them.

#### 7 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L184](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/e51d58999972a16065568487dbee797b572990b6/src/Engine/Hymn.cpp#L184): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L322](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/e51d58999972a16065568487dbee797b572990b6/src/Engine/Manager/RenderManager.cpp#L322): The function 'GetDirectionalLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L361](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/e51d58999972a16065568487dbee797b572990b6/src/Engine/Manager/RenderManager.cpp#L361): The function 'GetMaterials' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L379](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/e51d58999972a16065568487dbee797b572990b6/src/Engine/Manager/RenderManager.cpp#L379): The function 'GetMeshes' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L398](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/e51d58999972a16065568487dbee797b572990b6/src/Engine/Manager/RenderManager.cpp#L398): The function 'GetPointLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L419](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/e51d58999972a16065568487dbee797b572990b6/src/Engine/Manager/RenderManager.cpp#L419): The function 'GetSpotLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Engine/Manager/RenderManager.cpp (hunk 3 above)
Comment (LarsIngo) [reply src/Engine/Manager/RenderManager.cpp]: No, but it's (hopefully) correct

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 9 issues
* ![MINOR][MINOR] 9 minor

Watch the comments in this conversation to review them.

#### 7 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L184](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d2201a0a218843fbfd221484ada7caf3d8d48a1c/src/Engine/Hymn.cpp#L184): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L322](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d2201a0a218843fbfd221484ada7caf3d8d48a1c/src/Engine/Manager/RenderManager.cpp#L322): The function 'GetDirectionalLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L361](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d2201a0a218843fbfd221484ada7caf3d8d48a1c/src/Engine/Manager/RenderManager.cpp#L361): The function 'GetMaterials' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L379](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d2201a0a218843fbfd221484ada7caf3d8d48a1c/src/Engine/Manager/RenderManager.cpp#L379): The function 'GetMeshes' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L398](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d2201a0a218843fbfd221484ada7caf3d8d48a1c/src/Engine/Manager/RenderManager.cpp#L398): The function 'GetPointLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L419](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/d2201a0a218843fbfd221484ada7caf3d8d48a1c/src/Engine/Manager/RenderManager.cpp#L419): The function 'GetSpotLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 9 issues
* ![MINOR][MINOR] 9 minor

Watch the comments in this conversation to review them.

#### 7 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [Hymn.cpp#L184](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/e4943fef9fef5bc345ce3143ff5c61ba02746129/src/Engine/Hymn.cpp#L184): Variable 'model' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)
1. ![MINOR][MINOR] [RenderManager.cpp#L322](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/e4943fef9fef5bc345ce3143ff5c61ba02746129/src/Engine/Manager/RenderManager.cpp#L322): The function 'GetDirectionalLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L361](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/e4943fef9fef5bc345ce3143ff5c61ba02746129/src/Engine/Manager/RenderManager.cpp#L361): The function 'GetMaterials' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L379](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/e4943fef9fef5bc345ce3143ff5c61ba02746129/src/Engine/Manager/RenderManager.cpp#L379): The function 'GetMeshes' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L398](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/e4943fef9fef5bc345ce3143ff5c61ba02746129/src/Engine/Manager/RenderManager.cpp#L398): The function 'GetPointLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)
1. ![MINOR][MINOR] [RenderManager.cpp#L419](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/e4943fef9fef5bc345ce3143ff5c61ba02746129/src/Engine/Manager/RenderManager.cpp#L419): The function 'GetSpotLights' is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**